presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/serde/EsriGeometrySerde.java

```diff
@@ -171,7 +174,12 @@ public static OGCGeometry deserialize(Slice shape)
         verify(input.available() > 0);
         int length = input.available() - 1;
         GeometrySerializationType type = GeometrySerializationType.getForCode(input.readByte());
-        return readGeometry(input, shape, type, length);
+        try {
+            return readGeometry(input, shape, type, length);
+        }
+        catch (GeometryException e) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e.getMessage(), e);
```

Comment:
When I ran TestGeometrySerialization I didn't see this exercised in the debugger.